### ISSUE TITLE
Minified code on a javascript file, uglifies it

### ISSUE THREAD

**Issue by jarodium** [atom/atom#11593]
### Prerequisites

* [ ] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
* [ ] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [ ] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [ ] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [ ] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
* [yes] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

### Description

I am building a javascript file ( development file ) which has both minified ( Google closure compiler ) and un-minified code.
When I save the file, automatically unminifies the minified block.
### Steps to Reproduce

1. [First Step]
Take the following code

```js
function str_replace(h,d,b,f){var a=0,c=0,g="",l="",m=0,n=0,k=[].concat(h),e=[].concat(d),p="[object Array]"===Object.prototype.toString.call(e),q="[object Array]"===Object.prototype.toString.call(b);b=[].concat(b);if("object"===typeof h&&"string"===typeof d){g=d;d=[];for(a=0;a<h.length;a+=1)d[a]=g;e=[].concat(d);p="[object Array]"===Object.prototype.toString.call(e)}f&&(this.window[f]=0);a=0;for(m=b.length;a<m;a++)if(""!==b[a])for(c=0,n=k.length;c<n;c++)g=b[a]+"",l=p?void 0!==e[c]?e[c]:"":e[0],b[a]=
g.split(k[c]).join(l),f&&(this.window[f]+=g.split(k[c]).length-1);return q?b:b[0]};

function _aux_identLocal(id) {
    for (i = 0; i < LOCAIS.length; i++) {
        if (LOCAIS[i].id == id) {
            return LOCAIS[i].nome;
        }
    }
}
```
2. [Second Step]
Save the file

**Expected behavior:** [What you expected to happen]
I would expect the code to be:

```js
function str_replace(h,d,b,f){var a=0,c=0,g="",l="",m=0,n=0,k=[].concat(h),e=[].concat(d),p="[object Array]"===Object.prototype.toString.call(e),q="[object Array]"===Object.prototype.toString.call(b);b=[].concat(b);if("object"===typeof h&&"string"===typeof d){g=d;d=[];for(a=0;a<h.length;a+=1)d[a]=g;e=[].concat(d);p="[object Array]"===Object.prototype.toString.call(e)}f&&(this.window[f]=0);a=0;for(m=b.length;a<m;a++)if(""!==b[a])for(c=0,n=k.length;c<n;c++)g=b[a]+"",l=p?void 0!==e[c]?e[c]:"":e[0],b[a]=
g.split(k[c]).join(l),f&&(this.window[f]+=g.split(k[c]).length-1);return q?b:b[0]};

function _aux_identLocal(id) {
    for (i = 0; i < LOCAIS.length; i++) {
        if (LOCAIS[i].id == id) {
            return LOCAIS[i].nome;
        }
    }
}
```
**Actual behavior:** [What actually happened]
```js
function str_replace(h, d, b, f) {
    var a = 0,
        c = 0,
        g = "",
        l = "",
        m = 0,
        n = 0,
        k = [].concat(h),
        e = [].concat(d),
        p = "[object Array]" === Object.prototype.toString.call(e),
        q = "[object Array]" === Object.prototype.toString.call(b);
    b = [].concat(b);
    if ("object" === typeof h && "string" === typeof d) {
        g = d;
        d = [];
        for (a = 0; a < h.length; a += 1) d[a] = g;
        e = [].concat(d);
        p = "[object Array]" === Object.prototype.toString.call(e)
    }
    f && (this.window[f] = 0);
    a = 0;
    for (m = b.length; a < m; a++)
        if ("" !== b[a])
            for (c = 0, n = k.length; c < n; c++) g = b[a] + "", l = p ? void 0 !== e[c] ? e[c] : "" : e[0], b[a] =
                g.split(k[c]).join(l), f && (this.window[f] += g.split(k[c]).length - 1);
    return q ? b : b[0]
};

function _aux_identLocal(id) {
    for (i = 0; i < LOCAIS.length; i++) {
        if (LOCAIS[i].id == id) {
            return LOCAIS[i].nome;
        }
    }
}
```
### Versions
Atom version: 1.6.2
OSX Version: 10.11.4 (15E65)

**Comment by 50Wliu** [comment]
Please make sure you can reproduce this behavior in safe mode.  If you can just comment here and I'll reopen this issue.

(Also, the latest Atom version is currently 1.7.2)

**Comment by jarodium** [comment]
Hello @50Wliu 

I have upgraded Atom to 1.7.2 and boot it up to safe mode.
Created a new js file, and saved it with the minified version. The issue does not occur.

Then I booted Atom regularly and it happens. 
Can it be one of the default packages that is behind this issue?

Thank you for the time
Regards

**Comment by 50Wliu** [comment]
@jarodium It's most likely one of your installed packages causing this issue.  You can try using the [package-cop](https://atom.io/packages/package-cop) package to figure out exactly which one is behind your issue.  Off the top of my head, if you have `atom-beautify` installed, you may want to try disabling that first.

**Comment by jarodium** [comment]
@50Wliu i believed I have sorted this one out.
When I created a js file with one minified function, Atom did not triggered the issue, but, for insanity purproses only, when I pasted the same function below that one, it immediately triggered the issue. 
I suspect the formatter ( i have no beautify or minify packages. Only tidy and json formatters. ), no idea which one, is confused with both un-minified code and minified code and expands. Also, when saving two minified functions, it expands the code. When using one minified function, it does not expand it.

Regards
